### PR TITLE
[Quantum] Ship peer dependencies in vendored sdks folder (25369)

### PR DESCRIPTION
**Pull request by warren-jones**
**Original description from User Story 25369:**

> Azure CLI team:
> 
> Could you help vendor(copy) the Azure-related packages to the quantum extensions' vendor directory?
> 
> We require that all Azure-related packages be directly copied to the vendor directory; otherwise, there might be conflicts with the dependencies in the main repo. 
> 
> This may cause users to be unable to use the extension, and may also cause the automatic update of the extension documentation to fail.
> 
> The quantum extension has directly set azure-storage-blob as a dependency: [link to the setup.py file](https://github.com/Azure/azure-cli-extensions/blob/main/src/quantum/setup.py#L37).
> 
> This is the [PR](https://github.com/Azure/azure-cli-extensions/pull/7898) submitted by the k8s-runtime team. You can refer to it to make the necessary modifications.
> 

**Changes for this PR:**

- Copy the [azure.storage.blob](https://pypi.org/project/azure-storage-blob/) files to `src\quantum\azext_quantum\vendored_sdks\azure_storage_blob`
- Modify the associated  `from ... import` statement in `src\quantum\azext_quantum\_storage.py`
- Remove the `DEPENDENCIES` statement in `src\quantum\setup.py`
- Create a wiki page to explain how to update `vendored_sdks\azure_storage_blob` in the future.

New wiki: [How to update the Azure storage blob client](https://dev.azure.com/ms-azurequantum/AzureQuantum/_wiki/wikis/Azure%20Quantum%20wiki/1324/How-to-update-the-Azure-storage-blob-client)

No changes will be perceived by customers.
A Quantum CLI extension release will follow this PR in a subsequent Release PR.

---

### Related commands
`az quantum run`
`az quantum execute`
`az quantum job submit`

---

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

### CODE REVIEW

**Comment by yonzhan** [comment]
Quantum

File: src/quantum/eng/Generate-Vendored-SDKs.ps1 [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        if ($ClientToGenerate -eq "storage-blob")
+        {
+            $AutoRestConfig = "$SwaggerRepoUrl/blob/$SwaggerRepoBranch/specification/storage/data-plane/Microsoft.BlobStorage/readme.md"
+            $OutputFolder = "./azext_quantum/vendored_sdks/azure_storage_blob/"
+        }
+        else
+        {
+            Write-Error "ERROR: ClientToGenerate parameter not recognized."
+            return -1
+        }
```
Comment (kikomiss) [inline src/quantum/eng/Generate-Vendored-SDKs.ps1:82]: Do we still generate the blob client this way? From the wiki I understood that we're copying it over from PyPi package as is. If this is not currently used, might be we could remove this code to not confuse ourselves later on how to get vendored_sdk for `azure_storage_blob`

Comment (warren-jones) [reply src/quantum/eng/Generate-Vendored-SDKs.ps1:82]: No, we will not generate the client files this way.  I intended to delete this script.  Deleting it now.

**Review by kikomiss — APPROVED** [review]

**Comment by warren-jones** [comment]
HISTORY.rst and setup.py updates will accompany an upcoming release PR as is customary in our extension-release workflow.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**